From 7d2857b7d9a45aa5f7cfe93f61258152b0dcae07 Mon Sep 17 00:00:00 2001
From: "Torne (Richard Coles)" <torne@google.com>
Date: Tue, 3 Nov 2020 19:02:16 +0000
Subject: [PATCH 01/73] android: upstream missed manifest changes for Q/R.

On Q+ we disabled audio capture as we have no way to selectively disable
it for incognito tabs, but this was missed when upstreaming Q.

Chrome needs QUERY_ALL_PACKAGES permission on R+ to check for any
possible intent handler for URLs. This was added downstream but missed
while upstreaming R support; add it now.

Bug: 948282, 1042023
Change-Id: I4136beddcba0227ef6cfc2073f90d645151ce6bb
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/2507575
Reviewed-by: David Trainor <dtrainor@chromium.org>
Reviewed-by: Theresa  <twellington@chromium.org>
Commit-Queue: Richard Coles <torne@chromium.org>
Cr-Commit-Position: refs/heads/master@{#823642}
---
 .../monochrome_public_bundle.AndroidManifest.expected           | 2 ++
 .../trichrome_chrome_bundle.AndroidManifest.expected            | 2 ++
 chrome/android/java/AndroidManifest.xml                         | 2 ++
 3 files changed, 6 insertions(+)

diff --git a/chrome/android/expectations/monochrome_public_bundle.AndroidManifest.expected b/chrome/android/expectations/monochrome_public_bundle.AndroidManifest.expected
index 84b9915d3b2b..b2a649244593 100644
--- a/chrome/android/expectations/monochrome_public_bundle.AndroidManifest.expected
+++ b/chrome/android/expectations/monochrome_public_bundle.AndroidManifest.expected
@@ -37,6 +37,7 @@
   <uses-permission android:name="android.permission.MANAGE_ACCOUNTS"/>
   <uses-permission android:name="android.permission.MODIFY_AUDIO_SETTINGS"/>
   <uses-permission android:name="android.permission.NFC"/>
+  <uses-permission android:name="android.permission.QUERY_ALL_PACKAGES"/>
   <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE"/>
   <uses-permission android:name="android.permission.READ_SYNC_SETTINGS"/>
   <uses-permission android:name="android.permission.READ_SYNC_STATS"/>
@@ -63,6 +64,7 @@
   <uses-permission-sdk-23 android:name="android.permission.USE_FINGERPRINT"/>
   <uses-sdk android:minSdkVersion="24" android:targetSdkVersion="30"/>
   <application
+      android:allowAudioPlaybackCapture="false"
       android:allowBackup="false"
       android:extractNativeLibs="false"
       android:icon="@drawable/ic_launcher"
diff --git a/chrome/android/expectations/trichrome_chrome_bundle.AndroidManifest.expected b/chrome/android/expectations/trichrome_chrome_bundle.AndroidManifest.expected
index 6c537f8d034c..b58681054c7a 100644
--- a/chrome/android/expectations/trichrome_chrome_bundle.AndroidManifest.expected
+++ b/chrome/android/expectations/trichrome_chrome_bundle.AndroidManifest.expected
@@ -37,6 +37,7 @@
   <uses-permission android:name="android.permission.MANAGE_ACCOUNTS"/>
   <uses-permission android:name="android.permission.MODIFY_AUDIO_SETTINGS"/>
   <uses-permission android:name="android.permission.NFC"/>
+  <uses-permission android:name="android.permission.QUERY_ALL_PACKAGES"/>
   <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE"/>
   <uses-permission android:name="android.permission.READ_SYNC_SETTINGS"/>
   <uses-permission android:name="android.permission.READ_SYNC_STATS"/>
@@ -63,6 +64,7 @@
   <uses-permission-sdk-23 android:name="android.permission.USE_FINGERPRINT"/>
   <uses-sdk android:minSdkVersion="29" android:targetSdkVersion="30"/>
   <application
+      android:allowAudioPlaybackCapture="false"
       android:allowBackup="false"
       android:extractNativeLibs="false"
       android:icon="@drawable/ic_launcher"
diff --git a/chrome/android/java/AndroidManifest.xml b/chrome/android/java/AndroidManifest.xml
index 4b942ff06700..a19e52ce60ef 100644
--- a/chrome/android/java/AndroidManifest.xml
+++ b/chrome/android/java/AndroidManifest.xml
@@ -53,6 +53,7 @@ by a child template that "extends" this file.
     <uses-permission android:name="android.permission.MANAGE_ACCOUNTS"/>
     <uses-permission android:name="android.permission.MODIFY_AUDIO_SETTINGS"/>
     <uses-permission android:name="android.permission.NFC"/>
+    <uses-permission android:name="android.permission.QUERY_ALL_PACKAGES" />
     <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE"/>
     <uses-permission android:name="android.permission.READ_SYNC_SETTINGS"/>
     <uses-permission android:name="android.permission.READ_SYNC_STATS"/>
@@ -174,6 +175,7 @@ by a child template that "extends" this file.
         android:allowBackup="false"
         {% endif %}
         android:networkSecurityConfig="@xml/network_security_config"
+        android:allowAudioPlaybackCapture="false"
         {% block extra_application_attributes %}{% endblock %}>
 
       {% macro application_definitions() %}
-- 
2.29.2

